### PR TITLE
Add support for CMCD-Request header nor field to prefetch plugin

### PR DESCRIPTION
**Pull request by traeak**
This adds support to the prefetch plugin for use with the Cmcd-Request header with nor field relative path.

### CODE REVIEW

File: plugins/prefetch/plugin.cc [hunk 1]
```diff
@@ -35,7 +35,7 @@
 #include "headers.h"
 #include "evaluate.h"
 
-static const char *
+const char *
```
Comment (cmcfarlen) [inline plugins/prefetch/plugin.cc:38]: This can be static

File: plugins/prefetch/plugin.cc [hunk 2]
```diff
@@ -450,7 +499,7 @@ contHandleFetch(const TSCont contp, TSEvent event, void *edata)
     }
     if (!appendCacheKey(txnp, reqBuffer, data->_cachekey)) {
       PrefetchError("failed to get the cache key");
-      TSHttpTxnReenable(txnp, TS_EVENT_HTTP_ERROR);
+      TSHttpTxnReenable(txnp, TS_EVENT_HTTP_CONTINUE);
```
Comment (cmcfarlen) [inline plugins/prefetch/plugin.cc:502]: This might be a surprising change to existing users.  Can you explain why this is necessary?

Comment (traeak) [reply plugins/prefetch/plugin.cc:502]: A failed prefetch was causing the original transaction to abort.  I'm sure that's undesireable behavior.

**Comment by traeak** [comment]
[approve ci fedora]

**Comment by traeak** [comment]
[approve ci autest]

**Comment by traeak** [comment]
Adding some extra checks to ensure that the specified relative path is "simple" and not malformed.

File: plugins/prefetch/plugin.cc (hunk 2 above)
Comment (traeak) [reply plugins/prefetch/plugin.cc]: i just restored this.

**Comment by traeak** [comment]
[approve ci autest]

**Comment by traeak** [comment]
[approve ci autest]

File: plugins/prefetch/plugin.cc [hunk 3]
```diff
@@ -460,7 +542,7 @@ contHandleFetch(const TSCont contp, TSEvent event, void *edata)
         /* first-pass */
         if (!config.isExactMatch()) {
           data->_fetchable = state->acquire(data->_cachekey);
-          PrefetchDebug("request is %s fetchable", data->_fetchable ? " " : " not ");
+          PrefetchDebug("request is %s fetchable", data->_fetchable ? "" : "not");
```
Comment (maskit) [inline plugins/prefetch/plugin.cc:545]: More appropriate fix would be removing spaces around %s instead of changing the literals.

Comment (traeak) [reply plugins/prefetch/plugin.cc]: reverted the change

**Review by cmcfarlen — APPROVED** [review]

**Pull request merged**